evm/logic/system.py

```diff
@@ -52,7 +74,7 @@ def create(computation):
 
     call_data = computation.memory.read(start_position, size)
 
-    create_msg_gas = computation.gas_meter.gas_remaining
+    create_msg_gas = max_child_gas_modifier(computation.gas_meter.gas_remaining)
```

Comment:
This doesn't sit well with me.  Functions which pass around methods strike me as harder to follow than using a class.  Do you think there is a cleaner way to do this with a class abstraction similar to how it's done with the `Call` classes?